### DISCUSSION TITLE
요소 높이 설정 관련 안내

### DISCUSSION THREAD

**Discussion by July249** [beyondDevelops/AnimalTalk#74] [Announcements]
전체 레이아웃과 내부 콘텐츠 요소의 높이를 설정하는 것에 대한 안내사항입니다.

- 모든 콘텐츠가 들어가는 레이아웃 전반에는 디바이스 및 브라우저 창 크기에 따라 높이가 변경될 수 있도록 100vh를 제공합니다. 따라서 .page 에는 h-screen 클래스명을 제공합니다.

- main 요소 자체에는 basis-full을 제공하여 main 콘텐츠만으로 뷰포트 height 100%을 차지하지 못할 경우를 대비하여 유휴공간을 콘텐츠로 인식할 수 있도록 합니다. 이는 header와 footer를 사용자 웹 브라우저 창의 맨 위와 맨 아래에 고정시키기 위함입니다.

- main 요소 안의 내부 콘텐츠는 기본적으로 main 요소 안의 상단에 위치를 잡아 주는 특징이 있습니다. 따라서 별도로 height를 지정할 필요는 없습니다. (내부 콘텐츠 사이즈에 맞게 유연한 높이를 가지도록 하기 위함입니다.) 하지만 필요에 따라 mb-auto를 제공하여 줄 수 있습니다.

**Comment by IntHyun** [comment]
고생하셨습니다.
